pup-0002.md

```diff
@@ -0,0 +1,121 @@
+[Comments section](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+of the Google Python Style Guide. Exceptions and clarifications will be explicitly stated in the
+Pulp style guide.
+
+### Exceptions and Clarifications
+
+1. Modules should not include license information.
+
+2. The type of each `Arg` value should be included after the variable name in parentheses. The type
+of each `Returns` value should be the first item on the line.
+
+3. Following the type of `Arg` and `Returns` values, there will be a colon and a single space
+followed by the description. Additional spaces should not be used to align types and descriptions.
+
+4. `Fields` and `Relations` sections will be used when documenting fields on Django models.
+The `Fields` section will be used for non-related fields on Model classes, the `Relations`
+section will be used for related fields on Model classes.
+
+### Example
+
+    The first section is a summary, which should be restricted to a single line.
+
+    More detailed information follows the summary after a blank line. This section can be as many
+    line as necessary.
+
+    Args:
+        arg1 (str): The argument is visible, and its type is clearly indicated.
+        much_longer_argument (str): Types and descriptions are not aligned.
+    Returns:
+        bool: The return value and type is very clearly visible.
+
+### Language
+
+"Google style docstrings" is the preferred way to refer to the style changes outlined here. It
+should be noted that the phrase "Napoleon style docstrings" was originally used. Using the term
+"Napoleon" is clear because it accepts docstrings of multiple styles.
+
+### Requirements
+
+Docstrings will be used for auto-documentation and must be parsable by the
+[Napoleon plugin for Sphinx.](http://www.sphinx-doc.org/en/stable/ext/napoleon.html)
+
+
+### Porting from Pulp 2.Y
+
+The 3.0 development guide will be updated to state that all ported code should contain updated
+docstrings. Regular expressions to ease large changes will be included there.
+
+### Putting It Into Practice
+
+Upon acceptance, the following actions will be taken:
+
+1. The Pulp style guide will be updated to include Google style docstrings.
+2. Docs will be built with the Napoleon plugin which supports this style and the standard
+Sphinx style.
+3. The [developer wiki](https://pulp.plan.io/projects/pulp/wiki/#Pulp-30-development) will be
+updated to include docstring porting guidelines.
+
+
+## Alternative: Use Standard Sphinx Style Docstrings (Keep Current Style)
+
+Pros:
+  - 2.Y docstrings could be moved without modification.
+  - The Google Python Docstring Style may not be widely adopted and understood.
+
+Cons:
+  - Standard Sphinx docstrings are not very readable for humans.
+  - Standard Sphinx docstrings contain redundant words like `param`.
+  - The `type` of each param takes up its own line.
+  - Pulp 2.Y docstrings are inconsistent and vary in their use of `:param` and `@param`.
```

Comment:
This is a relic of the last time we changed styles. It is not part of our current style, so I think this point should be removed.

Pulp previously used the epydoc style: http://epydoc.sourceforge.net/

When we changed to reST, we decided to convert old code on a rolling basis, as it was worked on. As a result, some code that hasn't been touched since then is still in the old style.

At this point, converting epydoc to reST would probably be less effort than converting it to the Google style, which is why I don't think this is a "con".